New Community page, as designed by @alexcontini

cjyabraham opened the pull request:
This new Community page was designed over the last few months by @alexcontini.  I helped implement the Upcoming Events part of it.

cjyabraham commented:
CLA now signed.
/assign @jaredbhatti

jaredbhatti commented:
Thanks, I was pinged by acontini to hold off on approval until he pings me with a few updates.

rlenferink commented on static/css/newcommunity.css:
```diff
@@ -0,0 +1,777 @@
+.banner1 {
+position:relative;
+float:left;
+
```
```suggestion
width: 100%;
```

Maybe it will be good to add a width of 100% to the `.banner1` as well. Otherwise the image can't be full width.

rlenferink commented:
What is the status on this? As I really like the newly designed community page :D

alexcontini commented:
@parispittman @castrojo this is the PR for the new community page (test page here - https://deploy-preview-13046--kubernetes-io-master-staging.netlify.com/community/)

We are still waiting on some things before merging right? just wanted to double check!

mistyhacks commented:
/hold

(until the waiting-for-things are done)

alexcontini commented:
@parispittman @castrojo here is the updated preview page with Paris' image and text updates - https://deploy-preview-13046--kubernetes-io-master-staging.netlify.com/community/

let me know if all looks good when you have a chance!

parispittman commented:
some folks are reporting that the image fencing with the descriptions of our comms channels are in and out of the box with resizing of web browser. 

also - the code of conduct, videos, etc toolbar is not working for some folks

![Screen Shot 2019-04-15 at 9 24 06 AM](https://user-images.githubusercontent.com/5912992/56136767-43b91d80-5f48-11e9-8584-ecca48b0c03a.png)

jonasrosland commented:
Same comment as @parispittman above, the scaling is a little off in those boxes one larger screens.

For the events, can we organize the events so that it's spread out across multiple lines instead of being just one line?
Going from this:
![image](https://user-images.githubusercontent.com/1690215/56303085-2aab9a80-6109-11e9-9453-586242d8da25.png)
to
```
Event
Date
Location
```

After clicking the links in the toolbar, the drop-down menu obscures the titles:

![image](https://user-images.githubusercontent.com/1690215/56303350-b8878580-6109-11e9-8e79-eab9d3528232.png)

![image](https://user-images.githubusercontent.com/1690215/56303401-d2c16380-6109-11e9-960a-21040cca148e.png)


![image](https://user-images.githubusercontent.com/1690215/56303371-c0dfc080-6109-11e9-90d0-3b0ae8325947.png)

![image](https://user-images.githubusercontent.com/1690215/56303388-cc32ec00-6109-11e9-8564-03a1da7119ad.png)

jonasrosland commented on content/en/community/_index.html:
```diff
@@ -1,69 +1,243 @@
+<div class="intro">
+
+<p>The Kubernetes community -- users, contributors, and the culture we've built together -- is one of the biggest reasons for the meteoric rise of this open source project. Our culture and values continue to grow and change as the project itself grows and changes. We all work together toward constant improvement of the project and the ways we work on it.
+<br><br>We are the people who file issues and pull requests, attend SIG meetings, Kubernetes meetups, and KubeCon, advocate for it's adoption and innovation, run <code>kubectrl get pods</code>, and contribute in a thousand other vital ways. Read on to learn how you can get involved and become part of this amazing&nbsp;community.</p>
```
This should be `kubectl get pods`, right?
And I think we can remove `&nbsp;` on the last line.

jonasrosland commented on content/en/community/_index.html:
```diff
@@ -1,69 +1,243 @@
+
+<h1 style="margin-top:0px">Videos</h1>
+
+<div style="margin-bottom:4%;font-weight:300;text-align:center;padding-left:10%;padding-right:10%">We're on youtube, a lot. Subscribe for a wide range of&nbsp;topics.</div>
```
Should be `YouTube`, and I think we can remove the `&nbsp;` here too.

alexcontini commented:
@jonasrosland @parispittman @mrbobbytables 

I have made the fixes, when you have a chance, will you please take a look at https://deploy-preview-13046--kubernetes-io-master-staging.netlify.com/community/ and let me know if all looks good now.

The only one I was unsure about it the anchor links, they were working for me in both browsers so it was hard to figure out the issue, i did change something though so let me know if not fixed now. I also moved up the anchors so they shouldn't cut off the titles anymore hopefully

mrbobbytables commented:
@alexcontini That fixed it for me :) 
Thank you!

jonasrosland commented:
Hi @alexcontini!

Thank you for working on this, it's looking really good :)

The site still says `kubectrl get pods`, it should be `kubectl get pods`.

The squares at the bottom aren't uniform, but the text alignment looks much better :)

Large screen:
![image](https://user-images.githubusercontent.com/1690215/56591162-39d19300-65b6-11e9-8c5b-8db5b418d9f2.png)

Smaller screen:
![image](https://user-images.githubusercontent.com/1690215/56591507-e01d9880-65b6-11e9-84b8-4209c18adb5e.png)

I see the events are now split up into
```
Event
Location - Time
```

It looks a little strange still:
![image](https://user-images.githubusercontent.com/1690215/56591249-61286000-65b6-11e9-9b97-807bd47cdcca.png)

Could it be set up on 3 lines without any `-`? Like this:
```
Event
Location
Time
```

The anchor links looks better but still seems to require a little trimming:
CoC:
![image](https://user-images.githubusercontent.com/1690215/56591569-fc213a00-65b6-11e9-90ef-0b2d3080a40a.png)

Videos:
![image](https://user-images.githubusercontent.com/1690215/56591605-06dbcf00-65b7-11e9-970b-639c231d2c99.png)

Discussions now have a big block above it compared to before:
![image](https://user-images.githubusercontent.com/1690215/56591645-1824db80-65b7-11e9-9286-7f40f62e103b.png)

Same with events:
![image](https://user-images.githubusercontent.com/1690215/56591691-2a067e80-65b7-11e9-8798-83dc953b876f.png)

This is in Safari on latest Mac OSX, and the same happens in Chrome.

mrbobbytables commented:
@jonasrosland The event data is sourced from an external calendar. It'd be fixing it there.
The anchor links look okay for me on chrome (`Version 73.0.3683.103 (Official Build) (64-bit)`) on OSX.

The boxes do shift size when resizing.

alexcontini commented:
@mrbobbytables  @jonasrosland 
I have made some fixes: https://deploy-preview-13046--kubernetes-io-master-staging.netlify.com/community/  

1. The resource box alignment - I added a min-height, but it's been tricky to make it match perfectly at every width because the text is such different lengths in each box (that is why I set the height before, but then it was getting cut off) - let me know if this looks any better to you!

2. The anchor links - I added more spacing above the titles at the width where the menu becomes the mobile menu, so the issue before with the words getting cut off shouldn't happen anymore, let me know otherwise.

3. The event section text - @mrbobbytables mentioned this above but it's all feeding through a google calendar so I can't edit the way the text looks. I will ask someone on the team if they can change that though

Let me know if this looks good to you when you have a chance, and thank you for all the feedback!

jonasrosland commented:
Looks really good @alexcontini ! Thanks for those fixes, seems like it works great now.
The calendar piece is more of a nice-to-have than anything else, so I think that could be worked on afterwards.

Looks like you might need to squash a few commits to get rid of the nags from the bot though :)

/lgtm

zacharysarah commented:
@cjyabraham @alexcontini This looks so good! ✨ 

Some feedback:

1. There's a merge conflict to resolve prior to merging.
2. Please squash your commits by rebasing.
3. This PR is old enough that the preview is still displaying 1.13 as the latest version. That's not a problem, since there's no merge conflict with the version selector. Rebasing should pull in the latest changes and update the preview.

cjyabraham commented:
I have squashed and rebased the commits.  The merge conflict brought in the "Talk to Us" section that you can [see on the live site](https://kubernetes.io/community/).  It is now poorly formatted and also seems to have hidden our embedded twitter widget.  [See preview](https://5cc2ac31b59fc500094dbf75--kubernetes-io-master-staging.netlify.com/community/).

@alexcontini can you fix this please?

alexcontini commented:
@zacharysarah thank you! above issue is fixed (https://deploy-preview-13046--kubernetes-io-master-staging.netlify.com/community/) and all else should be ready now, are you able to merge when you have a chance?

parispittman commented:
Thanks @alexcontini and @cjyabraham for all of your hard work :)

zacharysarah commented:
/approve

zacharysarah commented:
/lgtm

@alexcontini `/hold cancel` when ready to merge.

alexcontini commented:
/hold cancel
